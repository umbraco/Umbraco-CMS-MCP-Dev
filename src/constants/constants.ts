export const BLANK_UUID = "00000000-0000-0000-0000-000000000000";
export const ROOT_DOCUMENT_TYPE_ID = "a95360e8-ff04-40b1-8f46-7aa4b5983096";
export const CONTENT_DOCUMENT_TYPE_ID = "b871f83c-2395-4894-be0f-5422c1a71e48";
export const Default_Memeber_TYPE_ID = "d59be02f-1df9-4228-aa1e-01917d806cda";
export const TextString_DATA_TYPE_ID = "0cc0eba1-9960-42c9-bf9b-60e150b429ae";
export const IMAGE_MEDIA_TYPE_ID = "cc07b313-0843-4aa8-bbda-871c8da728c8";
export const FOLDER_MEDIA_TYPE_ID = "f38bd2d7-65d0-48e6-95dc-87ce06ec2d3d";
export const EXAMPLE_IMAGE_PATH =
  "/src/umb-management-api/tools/temporary-file/__tests__/helpers/example.jpg";
