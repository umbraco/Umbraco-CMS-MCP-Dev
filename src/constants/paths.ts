import { join } from 'path';

// Use absolute path from current working directory
export const RESOURCES_DIR = join(process.cwd(), 'build', 'resources').replace(/\\/g, '/');