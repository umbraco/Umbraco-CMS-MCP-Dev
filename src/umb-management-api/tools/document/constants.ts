export const UmbracoDocumentPermissions = {
    Create: "Umb.Document.Create",
    Read: "Umb.Document.Read",
    Update: "Umb.Document.Update",
    Delete: "Umb.Document.Delete",
    Publish: "Umb.Document.Publish",
    Unpublish: "Umb.Document.Unpublish",
    Move: "Umb.Document.Move",
    Sort: "Umb.Document.Sort",
    Duplicate: "Umb.Document.Duplicate",
    PublicAccess: "Umb.Document.PublicAccess",
    CultureAndHostnames: "Umb.Document.CultureAndHostnames",
    CreateBlueprint: "Umb.Document.CreateBlueprint",
}