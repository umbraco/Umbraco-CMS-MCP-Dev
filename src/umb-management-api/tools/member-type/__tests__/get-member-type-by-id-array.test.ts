import GetMemberTypesByIdArrayTool from "../get/get-member-type-by-id-array.js";
import { MemberTypeBuilder } from "./helpers/member-type-builder.js";
import { MemberTypeTestHelper } from "./helpers/member-type-helper.js";
import { jest } from "@jest/globals";
import { BLANK_UUID } from "@/constants/constants.js";

describe("get-member-types-by-id-array", () => {
  const TEST_MEMBER_TYPE_NAME = "_Test Item MemberType";
  const TEST_MEMBER_TYPE_NAME_2 = "_Test Item MemberType2";
  let originalConsoleError: typeof console.error;

  // Helper to parse response, handling empty string as empty array
  const parseItems = (text: string) => {
    if (!text || text.trim() === "") return [];
    return JSON.parse(text);
  };

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
  });

  afterEach(async () => {
    console.error = originalConsoleError;
    await MemberTypeTestHelper.cleanup(TEST_MEMBER_TYPE_NAME);
    await MemberTypeTestHelper.cleanup(TEST_MEMBER_TYPE_NAME_2);
  });

  it("should get no member types for empty request", async () => {
    // Get all member types
    const result = await GetMemberTypesByIdArrayTool().handler(
      {},
      { signal: new AbortController().signal }
    );
    const items = parseItems(result.content[0].text as string);

    expect(items).toMatchSnapshot();
  });

  it("should get single member type by ID", async () => {
    // Create a member type
    const builder = await new MemberTypeBuilder()
      .withName(TEST_MEMBER_TYPE_NAME)
      .create();

    // Get by ID
    const result = await GetMemberTypesByIdArrayTool().handler(
      { id: [builder.getId()] },
      { signal: new AbortController().signal }
    );
    const items = parseItems(result.content[0].text as string);
    expect(items).toHaveLength(1);
    expect(items[0].name).toBe(TEST_MEMBER_TYPE_NAME);
    // Normalize for snapshot
    items[0].id = BLANK_UUID;
    expect(items).toMatchSnapshot();
  });

  it("should get multiple member types by ID", async () => {
    // Create first member type
    const builder1 = await new MemberTypeBuilder()
      .withName(TEST_MEMBER_TYPE_NAME)
      .create();

    // Create second member type
    const builder2 = await new MemberTypeBuilder()
      .withName(TEST_MEMBER_TYPE_NAME_2)
      .create();

    // Get by IDs
    const result = await GetMemberTypesByIdArrayTool().handler(
      {
        id: [builder1.getId(), builder2.getId()],
      },
      { signal: new AbortController().signal }
    );

    const items = parseItems(result.content[0].text as string);
    expect(items).toHaveLength(2);
    expect(items[0].name).toBe(TEST_MEMBER_TYPE_NAME);
    expect(items[1].name).toBe(TEST_MEMBER_TYPE_NAME_2);

    // Normalize for snapshot
    items.forEach((item: any) => {
      item.id = BLANK_UUID;
    });
    expect(items).toMatchSnapshot();
  });
});
