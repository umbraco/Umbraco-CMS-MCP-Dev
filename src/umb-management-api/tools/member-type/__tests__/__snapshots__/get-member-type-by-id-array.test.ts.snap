// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-member-types-by-id-array should get multiple member types by ID 1`] = `
[
  {
    "icon": "icon-user",
    "id": "00000000-0000-0000-0000-000000000000",
    "name": "_Test Item MemberType",
  },
  {
    "icon": "icon-user",
    "id": "00000000-0000-0000-0000-000000000000",
    "name": "_Test Item MemberType2",
  },
]
`;

exports[`get-member-types-by-id-array should get no member types for empty request 1`] = `[]`;

exports[`get-member-types-by-id-array should get single member type by ID 1`] = `
[
  {
    "icon": "icon-user",
    "id": "00000000-0000-0000-0000-000000000000",
    "name": "_Test Item MemberType",
  },
]
`;
